examples/asr/speech_to_text.py

```diff
@@ -38,62 +39,36 @@
 
 Override some args of optimizer:
     python speech_to_text.py \
-    AudioToTextDataLayer.manifest_filepath="./an4/train_manifest.json" \
-    AudioToTextDataLayer_eval.manifest_filepath="./an4/test_manifest.json" \
+    model.train_ds.manifest_filepath="./an4/train_manifest.json" \
+    model.validation_ds.manifest_filepath="./an4/test_manifest.json" \
     hydra.run.dir="." \
     pl.trainer.gpus=2 \
     pl.trainer.max_epochs=2 \
-    optim.args.params.betas=[0.8,0.5] \
-    optim.args.params.weight_decay=0.0001
+    model.optim.args.params.betas=[0.8,0.5] \
+    model.optim.args.params.weight_decay=0.0001
 
 Overide optimizer entirely
     python speech_to_text.py \
-    AudioToTextDataLayer.manifest_filepath="./an4/train_manifest.json" \
-    AudioToTextDataLayer_eval.manifest_filepath="./an4/test_manifest.json" \
+    model.train_ds.manifest_filepath="./an4/train_manifest.json" \
+    model.validation_ds.manifest_filepath="./an4/test_manifest.json" \
     hydra.run.dir="." \
     pl.trainer.gpus=2 \
     pl.trainer.max_epochs=2 \
-    optim.name=adamw \
-    optim.lr=0.001 \
-    ~optim.args \
-    +optim.args.betas=[0.8,0.5]\
-    +optim.args.weight_decay=0.0005
+    model.optim.name=adamw \
+    model.optim.lr=0.001 \
+    ~model.optim.args \
+    +model.optim.args.betas=[0.8,0.5]\
+    +model.optim.args.weight_decay=0.0005
 
 """
 
 
-@hydra.main(config_path="conf", config_name="config")
+@hydra.main(config_path="conf", config_name="config", strict=None)
 def main(cfg):
+    # omegaconf merg trainer stuff to optim - this is necessary to be able to correctly setup LR scheduler
+    cfg.model.optim.trainer = cfg.pl.trainer
```

Comment:
This should be `cfg.model.pl = cfg.pl`, please look below how we can use it then.

Reply:
ok